sdk/storage_blobs/tests/blob_operations.rs

```diff
@@ -0,0 +1,61 @@
+#![cfg(feature = "mock_transport_framework")]
+
+use azure_storage_blobs::container::PublicAccess;
+use azure_storage_blobs::prelude::{AsBlobServiceClient, AsContainerClient};
+use bytes::Bytes;
+use futures::StreamExt;
+use log::trace;
+
+mod setup;
+
+#[tokio::test]
+async fn put_block_blob_and_snapshot() {
+    const CONTAINER_NAME: &'static str = "test-container-put-block-blob-and-snapshot";
+    const BLOB_NAME: &'static str = "test-container-put-block-blob-and-snapshot";
+    let data = Bytes::from_static(b"abcdef");
+
+    let storage = setup::initialize("put_block_blob_and_snapshot").unwrap();
+    let blob_service = storage.blob_service_client();
+    let container = storage.container_client(CONTAINER_NAME);
+    let blob = container.blob_client(BLOB_NAME);
+
+    if blob_service
+        .list_containers()
+        .into_stream()
+        .next()
+        .await
+        .unwrap()
+        .unwrap()
+        .containers
+        .iter()
+        .find(|x| x.name == CONTAINER_NAME)
+        .is_none()
+    {
+        container
+            .create()
+            .public_access(PublicAccess::None)
+            .into_future()
+            .await
+            .unwrap();
+    }
+
+    // calculate md5 too!
+    let digest = md5::compute(&data[..]);
+
+    blob.put_block_blob(data)
+        .content_type("text/plain")
+        .hash(digest)
+        .into_future()
+        .await
+        .unwrap();
+
+    trace!("created {:?}", BLOB_NAME);
+
+    let snapshot = blob.snapshot().into_future().await.unwrap().snapshot;
+
+    trace!("crated snapshot: {:?} of {:?}", snapshot, BLOB_NAME);
```

Comment:
It would be nice to assert something here about the snapshot. Otherwise we're only testing that the deserialization doesn't outright fail.

Reply:
I'll think about this, but it is a bit tricky since you get back an "opaque" value that happens to be a timestamp. You could assert that the value is in the past, but then you wouldn't be treating it as opaque. I plan to add `list_snaphots` in the near future, which might be the better way to improve this test?